### PR TITLE
model checking support

### PR DESCRIPTION
**Pull request by coolya**
Implements #21 and is based on #25 hence #25 needs to be merged first. 

This pull request adds support to run the model checker directly from gradle on a project or subset of models from a project. It's configuration is closely aligned with the generator from command line part that we already have in the project.

In general the gradle part is not much different, a simple plugin with extensions for configuration and the a command line _backend_ that does the real work. Some of the decisions why to use certain classes are documented in the code directly but my main goal was to stay away from anything the uses the `ModelCheckerSetting` from MPS to avoid any interferences with the settings of the user.

Currently there is no way to configure which checks are executed and how the errors are reported. Everything is sent to log4j. In theory users could use the `log.xml` file in the MPS installation to customise the logging. Users could also use log4j [configuration](https://logging.apache.org/log4j/2.x/manual/configuration.html) via properties of env variables to redirect the log out put somewhere else. 

Customisation can be done by changing properties of the `checkmodels` task which is a [JavaExec](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/JavaExec.html) task.

## Future Work 

- Integrate with testing infrastructure (#30) 
- Better usage documentation with advanced examples

### CODE REVIEW

**Comment by arimer** [comment]
Is there a way how I can run the modelcheck test? After deploying the modelchecker locally with `modelcheck:publishPluginMavenPublicationToMavenLocal` I'll get an artifact with the version: `modelcheck-2018.2.5.2-SNAPSHOT`.

However, the build.gradle file for the **/test/modelcheck** project tries to use a modelcheck plugin with the following version:
` id("modelcheck") version "1.0-SNAPSHOT"` which it obviously can't find under the local repos.

Am I doing something wrong, or do we need to adapt the required migrationcheck version in the **/test/modelcheck** project?

File: README.md [hunk 1]
```diff
@@ -180,3 +180,74 @@ Parameters:
...
+* `plugins` - optional list of plugins to load before generation is attempted.
+  The notation is `new Plugin("someID", "somePath")`. Where the first parameter is the plugin id and the second the `short (folder) name`.
+* `pluginLocation` - location where to load the plugins from. Structure needs to be a flat folder structure similar to the
+  `plugins` directory inside of the MPS installation.
+* `models` - optional list of models to generate. If omitted all models in the project will be generated. Only full name
+  matched are supported and no RegEx or partial name matching.
+* `macros` - optional list of path macros. The notation is `new Macro("name", "value")`.
+* `projectLocation` - location of the MPS project to generate.
+* `errorNoFail` - report errors but do not fail the build.
+* `warningsAsError` - handles warnings as errors and will fail the build if any is found when `errorNoFail` is not set.
```
Comment (sergej-koscejev) [inline README.md]: There's an inconsistency between `--warning-as-error` and `warningsAsError` setting (the latter has an 's').

File: test/generate-build-solution/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+group = "test.de.itemis.mps.gradle.generate"
+version = "1.0-SNAPSHOT"
+
+plugins {
+    id("generate-models") version "1.0-SNAPSHOT"
```
Comment (sergej-koscejev) [inline test/generate-build-solution/build.gradle.kts]: This should now be 1.1-SNAPSHOT. @arimer is this maybe the problem that was preventing you from testing the plugin?

File: test/generate-simple/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+group = "test.de.itemis.mps.gradle.generate"
+version = "1.0-SNAPSHOT"
+
+plugins {
+    id("generate-models") version "1.0-SNAPSHOT"
```
Comment (sergej-koscejev) [inline test/generate-simple/build.gradle.kts]: Should be 1.1-SNAPSHOT as well

File: test/modelcheck/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import java.net.URI
+
+group = "test.de.itemis.mps.gradle.modelcheck"
+version = "1.0-SNAPSHOT"
+
+
+plugins {
+    id("modelcheck") version "1.0-SNAPSHOT"
```
Comment (sergej-koscejev) [inline test/modelcheck/build.gradle.kts]: And here too (1.1-SNAPSHOT)

**Pull request merged**